Schema/app/002-initial.blue.sql

```diff
@@ -104,33 +110,39 @@ EXECUTE PROCEDURE odin.app_owner_ledger_insert();
 -- APP USER
 CREATE TABLE odin.app_user (
     identity_id TEXT NOT NULL,
-    app_id TEXT NOT NULL,
-    state TEXT NOT NULL,
-    changed TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT NOW(),
-    pg_user TEXT NOT NULL,
-    PRIMARY KEY (identity_id, app_id),
     FOREIGN KEY (identity_id)
         REFERENCES odin.identity (id) MATCH SIMPLE
         ON UPDATE NO ACTION ON DELETE NO ACTION DEFERRABLE,
+
+    app_id TEXT NOT NULL,
     FOREIGN KEY (app_id)
         REFERENCES odin.app (app_id) MATCH SIMPLE
-        ON UPDATE NO ACTION ON DELETE NO ACTION DEFERRABLE
+        ON UPDATE NO ACTION ON DELETE NO ACTION DEFERRABLE,
+
+    PRIMARY KEY (identity_id, app_id),
+
+    state TEXT NOT NULL,
+    changed TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT NOW(),
+    pg_user TEXT NOT NULL
 );
 
 CREATE TABLE odin.app_user_ledger (
     reference TEXT NOT NULL,
     identity_id TEXT NOT NULL,
-    app_id TEXT NOT NULL,
-    state TEXT NOT NULL DEFAULT 'ACTIVE',
-    changed TIMESTAMP WITH TIME ZONE NOT NULL DEFAULT NOW(),
-    pg_user TEXT NOT NULL DEFAULT current_user,
-    PRIMARY KEY (reference, identity_id, app_id),
     FOREIGN KEY (identity_id)
         REFERENCES odin.identity (id) MATCH SIMPLE
         ON UPDATE NO ACTION ON DELETE NO ACTION DEFERRABLE,
+
+    app_id TEXT NOT NULL,
     FOREIGN KEY (app_id)
         REFERENCES odin.app (app_id) MATCH SIMPLE
-        ON UPDATE NO ACTION ON DELETE NO ACTION DEFERRABLE
+        ON UPDATE NO ACTION ON DELETE NO ACTION DEFERRABLE,
+
+    PRIMARY KEY (reference, identity_id, app_id),
+
+    state TEXT NOT NULL DEFAULT 'ACTIVE',
```

Comment:
Should there not be a constraint over these `state` columns with the allowed state values?

Reply:
I think for now it's OK because we are not sure about the choices of `state` yet.